src/writer/app_runner.py

```
@@ -467,17 +467,69 @@ def _execute_user_code(self) -> None:
+            use_stdout_redirect([
+                lambda entry: writer.core.initial_state.add_log_entry("info", "Stdout message during initialization", entry),
+                lambda entry: init_stdout_container.__setitem__(0, entry)
+            ]),
+            use_logging_redirect([
+                lambda entry: writer.core.initial_state.add_log_entry("info", "Logs during initialization", entry),
+                lambda entry: init_logs_container.__setitem__(0, entry)
+            ]),
         ):
             writeruserapp.__dict__["logger"] = user_code_logger
             code = compile(self.run_code, code_path, "exec")
             exec(code, writeruserapp.__dict__)
```

Ruff (0.14.5)

Use of `exec` detected

(S102)